trin-core/src/portalnet/storage.rs

```diff
@@ -0,0 +1,385 @@
+                DELETE_QUERY,
+                [key_to_remove_as_u64],
+            ).unwrap();
+            
+            match self.find_farthest() {
+                Err(e) => {
+                    error!("Failed to find farthest: {}", e);
+                },
+                Ok(farthest) => {
+                    self.farthest_key = Some(farthest.clone());
+                    self.data_radius = self.distance_to_key(&farthest);
+                }
+            }
+
+        } else {
+
+            let data_usage = self.get_total_storage_usage_kb();
+            if data_usage > self.storage_capacity_kb {
+              self.capacity_reached = true;
+            }
+
+        }
+
+        match self.farthest_key.as_ref() {
+            None => {
+                self.farthest_key = Some(key.to_string());
+            },
+            Some(farthest) => {
+                if self.distance_to_key(key) > self.distance_to_key(&farthest) {
+                    self.farthest_key = Some(key.clone());
+                }
+            }
+        }
+
+    }
+
+    pub fn get(&self, key: &String) -> Result<Option<Vec<u8>>, Error> {
+
+        self.db.get(key)
+
+    }
+
+    pub fn get_current_radius(&self) -> u64 {
+
+        self.data_radius
+
+    }
+
+    pub fn get_total_storage_usage_kb(&self) -> u64 {
+
+        let p: perf::MemoryUsageStats = perf::get_memory_usage_stats(Some(&[&self.db]), None)
+                                            .expect("Failed to get memory usage statistics.");
+
+        let db_total = self.get_total_size_of_directory_in_bytes(get_data_dir()).unwrap();
+
+        let total_in_bytes = p.mem_table_total + db_total;
+
+        ( total_in_bytes / 1000 ) as u64
+
+    }
+
+    pub fn find_farthest(&self) -> Result<String, String> {
+
+        let node_id_u64 = PortalStorage::byte_vector_to_u64(self.node_id.raw().to_vec());
+
+        println!("NODE ID: {}", node_id_u64);
+
+        let mut query = self.meta_db.prepare(
+            FIND_FARTHEST_QUERY,
+        ).unwrap();
+
+        let results = query.query_map([node_id_u64], |row| {
+            Ok(ContentKey {
+                key_long: row.get(0)?,
+            })
+        });
+
+        let x = results.unwrap().next().unwrap().unwrap().key_long;
+
+        Ok(x)
+
+    }
+
+    fn get_total_size_of_directory_in_bytes(&self, path: String) -> std::io::Result<u64> {
+
+        let metadata = fs::metadata(&path).unwrap();
+        let mut size = metadata.len();
+
+        if metadata.is_dir() {
+            for entry in fs::read_dir(&path)? {
+                let dir = entry?;
+                let path_string = dir.path().into_os_string().into_string();
+                size += self.get_total_size_of_directory_in_bytes(path_string.unwrap())?;
+            }
+        }
+
+        Ok(size)
+
+    }
+
+    pub fn distance_to_key(&self, key: &String) -> u64 {
+
+        let byte_vector = xor_two_values(
+            key.as_bytes(), &self.node_id.raw().to_vec()
```

Comment:
I'm not sure what `as_bytes()` does but, doesn't `key` need to be hashed to get the `content-id` here? Since I thought that the hash of the key was to be used in the distance fn.

Reply:
Yea this code doesn't yet account for the `key`-hash->`id` (where `id` is the actual key into the DB) thing. `key` anywhere in this currently refers to what we know as the `id`.  

Off the top of my head I think that this PortalStorage code should do that conversion itself, so that the code that uses it never has to know/care about content IDs. Just pass the key into `should_store(key)`, `store(key)`, and `get(key)`. 

`as_bytes()` is just turning the string into the `Vec<u8>` that the `xor_two_values` function expects.

Reply:
Are we using the `content_id` as the key into the db? It seemed to me that we are using the `content_key` as the actual index, since in protocol.rs in the main branch, that is what is being used as the index for DB queries (in the `FindContent`)

Reply:
i don't think that that code yet takes into account the fact the `content_id` is what's actually used as the key with which we index into the database.

I do think that the key/id nomenclature is a bit confusing because I think of a key as what's used to index a hash table, but in Portal Network the `id` is. Perhaps that's my own misunderstanding.